include/revng/ADT/Iterator.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <iterator>
+#include <type_traits>
+
+#include "llvm/ADT/STLExtras.h"
+
+auto derefereceIterator(auto Iter) {
```

Comment:
```suggestion
auto dereferenceIterator(auto Iter) {
```

Not sure this is worth having a file. If used in a single place, push it there. Otherwise, `STLExtras.h` sounds like a good place for these.